spec/controllers/api/v4/converts_controller_spec.rb

```diff
@@ -27,14 +27,51 @@
         end
       end
 
-    context 'unsupported format' do
-      let(:file) do
-        fixture_file_upload('files/llanfair')
+      context 'unsupported format' do
+        subject { post :create, params: {file: file, format: "llanfair" } }
+        let(:body) { JSON.parse(subject.body) }
+
+        it "returns a 400" do
+          expect(subject).to have_http_status 400
+        end
+
+        it 'returns an error body' do
+          expect(body['status']).to be_truthy
+          expect(body['message']).to be_truthy
+        end
       end
-      subject { post :create, params: {file: file, format: "llanfair" } }
 
-      it "returns a 400" do
-        expect(subject).to have_http_status 400
+      context "missing parameter" do
+        subject { post :create, params: {file: file} }
+        let(:body) { JSON.parse(subject.body) }
+
+        it "returns a 400" do
+          expect(subject).to have_http_status 400
+        end
+
+        it 'returns an error body' do
+          expect(body['status']).to be_truthy
+          expect(body['message']).to be_truthy
+        end
+      end
+    end
+
+    context 'from Malformed' do
```

Comment:
You don't need these double contexts, you can just do the one `'malformed file'` context at the same level as the Llanfair context.